File: 백준/Gold/1806. 부분합/부분합.py
N, S = map(int, input().split())
arr = list(map(int, input().split()))

l = 0
r = 0
total = 0
ans = float('inf')

while r < N:
    total += arr[r]
    r += 1

    while total >= S:
        ans = min(ans, r - l)
        total -= arr[l]
        l += 1

if ans == float('inf'):
    print(0)
else:
    print(ans)

File: 백준/Gold/1806. 부분합/부분합.cc
#include <iostream>
#include <algorithm>

using namespace std;

int N, S;
int arr[100000];
int ps[100001];

int main() {
    ios::sync_with_stdio(0);
    cin.tie(0);

    cin >> N >> S;

    ps[0] = 0;
    for (int i = 0; i < N; i++) {
        cin >> arr[i];
        ps[i + 1] = ps[i] + arr[i];
    }

    int left_idx = 0;
    int right_idx = 1;
    int ans = N + 1;

    while (left_idx <= right_idx && right_idx <= N) {
        int s = ps[right_idx] - ps[left_idx];

        if (s >= S) {
            ans = min(ans, right_idx - left_idx);
            left_idx++;
        } else {
            right_idx++;
        }
    }

    if (ans == N + 1) cout << 0;
    else cout << ans;

    return 0;
}
